packages/sdk/src/models/index.ts

```diff
@@ -208,6 +208,52 @@ export default class Models {
     return new Market(market as ExtendedMarketResponse, this.api);
   }
 
+  /**
+   * Fetches market_count variable from Zeitgeist chain
+   * This acts as a count of all markets which have been created, but includes those which have
+   * been cancelled, and all other statuses
+   */
+  async getMarketCount(): Promise<number | null> {
+    const count = (
+      await this.api.query.predictionMarkets.marketCount()
+    ).toJSON();
+    if (typeof count === 'number')
+      return count
+    else
+      // Spoiler: return null will never be run
+      return null    
```

Comment:
Why do we need a conditional check if the else branch won't run?

Reply:
`predictionMarkets.marketCount()` always returns a serialised `MarketId` so, even in edge cases if the its return Promise resolves, it will be with a stringified `MarketId`, which is always numeric (and difficult to see how it being numeric would ever change).
`.toJson()` may return a whole lot of types but in this case will always return a number. TS does not know that and so won't compile overlooking that possibility (and in fact, it's theoretically possible to change `MarketCount` to return, eg, a tuple which would no longer translate to number.
The `return null` case covers that next to impossibe case.